### PR TITLE
Issue #763 EhcacheActiveEntity unit tests and corrections

### PR DESCRIPTION
**Pull request by cljohnso**
This PR introduces unit tests for `EhcacheActiveEntity` and corrections for issues exposed by @lorban.

Additional unit tests to follow in additional PRs.

### CODE REVIEW

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java [hunk 1]
```diff
@@ -333,6 +379,17 @@ private EhcacheEntityResponse validate(ValidateCacheManager message) {
...
   private EhcacheEntityResponse createServerStore(ClientDescriptor clientDescriptor, CreateServerStore createServerStore) {
+    ClientState clientState = this.clientStateMap.get(clientDescriptor);
+    if (clientState == null) {
+      return failure(new IllegalStateException("Client " + clientDescriptor + " is not connected to the Clustered Store Manager"));
+    }
+    if (!clientState.isAttached()) {
+      return failure(new IllegalStateException("Client " + clientDescriptor + " is not attached to the Clustered Store Manager"));
+    }
+    if (!isConfigured()) {
+      return failure(new IllegalStateException("Clustered Store Manager is not configured"));
```
Comment (GaryWKeim) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: Could you include the clientDescriptor in this message like, "while attempting to create server-side storage for client {}"

Comment (cljohnso) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: This exception is actually transmitted back to the requesting client making the client ID somewhat redundant.  (It really didn't need to be provided in the exception messages above it.)  We should probably log these in the server before returning them and the client ID _needs_ to be in the log messages.

In general, we need  to "normalize" server- and client-side logging and exception messages to include information needed to track fault contributions back to clients (in particular).

I'll adjust this one (and the others like it) as an interim move until we take care of the more general logging/exception issue.

Comment (cljohnso) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: @ljacomet @chrisdennis @lorban  Opinion requested ...

It's clear that the client needs to be identified when logging -- at least server-side.  Is there any benefit to identifying the client in exception detail messages in exceptions that are sent back to the client?  Right now, some messages identify the client and some do not.  We need to pick something and normalized on it.

**Comment by cljohnso** [comment]
Rebased ...

**Comment by cljohnso** [comment]
Rebased with adjustments for PR #1050

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java (hunk 1 above)
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: I recommend opening an issue for this topic and having the discussion there.

Comment (cljohnso) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: Opened issue #1067.

**Pull request merged**